contentcuration/contentcuration/utils/publish.py

```diff
@@ -319,20 +323,43 @@ def recurse_nodes(self, node, inherited_fields):  # noqa C901
             )
 
             if node.kind_id == content_kinds.EXERCISE:
-                exercise_data = process_assessment_metadata(node, kolibrinode)
+                exercise_data = process_assessment_metadata(node)
```

Comment:
We defer generation of the kolibri AssessmentMetadata object until after file generation has completed.